.github/workflows/docker-build-and-publish.yaml

```diff
@@ -0,0 +1,86 @@
+name: Docker build and publish
+# Reusable workflow to build and push an image from a Dockerfile to any container registry
+
+on:
+  workflow_call:
+    inputs:
+      docker-context:
+        description: "The docker context."
+        required: false
+        default: "."
+        type: string
+      dockerfile-path:
+        description: "Path to the Dockerfile."
+        required: false
+        default: "Dockerfile"
+        type: string
+      docker-registry:
+        description: "Host where the image should be pushed to."
+        required: true
+        type: string
+      image-artifact-name:
+        description: "Name of the artifact that contains the Docker image.tar file to push, see https://github.com/actions/upload-artifact (Default is 'image-artifact')."
+        required: false
+        default: "image-artifact"
+        type: string
+      image-name:
+        description: "Name of Docker image (Default is the repository name)."
+        required: false
+        default: "${{ github.event.repository.name }}"
+        type: string
+      ref:
+        description: "Ref to checkout"
+        required: false
+        default: ""
+        type: string
+      working-directory:
+        description: "Working directory for your Docker artifacts. (Default is .)"
+        required: false
+        default: "."
+        type: string
+    secrets:
+      docker-publisher:
+        description: "Publisher to prefix Docker image (e.g. 'my-publisher')."
+        required: true
+      docker-user:
+        description: "Username for the Docker registry login."
+        required: true
+      docker-password:
+        description: "Password for the Docker registry login."
+        required: true
+      # if this workflow should be automatically triggered after a new tag was pushed by another action, then configure the github token to overcome it (as described here: https://github.com/orgs/community/discussions/27028)
+      github-token:
+        description: "GitHub token."
+        required: false
```

Comment:
This needs a default then as well

Reply:
i set a default value. but i had to move from the `secrets` block since secrets don't have a `default` property

Reply:
Ok, that's not ideal, but we have no other choice here I guess

Reply:
Don't forget to also change the `secret.` to a `input.` like the failing workflow is telling you